src/inbox/mod_inbox_utils.erl

```diff
@@ -50,7 +50,9 @@ write_to_sender_inbox(Server, Sender, Receiver, Packet) ->
     RemoteBareJid = jid:to_binary(jid:to_bare(Receiver)),
     %% no unread for a user because he writes new messages which assumes he read all previous messages.
     Count = integer_to_binary(0),
-    ok = mod_inbox_backend:set_inbox(Username, Server, RemoteBareJid, Content, Count, MsgId).
+    Timestamp = erlang:timestamp(),
```

Comment:
Did you consider using `os:system_time/1` which would return integer in the specified unit? It would eliminate calls to `usec:from_now`.